app/command_autocomplete_test.go

```diff
@@ -609,3 +611,66 @@ func createJiraAutocompleteData() *model.AutocompleteData {
 
 	return jira
 }
+
+func TestDynamicListArgsForBuiltin(t *testing.T) {
+	th := Setup(t)
+	defer th.TearDown()
+
+	provider := &testProvider{}
+	RegisterCommandProvider(provider)
+
+	command := provider.GetCommand(th.App, nil)
+	emptyCmdArgs := &model.CommandArgs{}
+
+	t.Run("GetAutoCompleteListItems", func(t *testing.T) {
+		suggestions := th.App.getSuggestions(emptyCmdArgs, []*model.AutocompleteData{command.AutocompleteData}, "", "bogus --dynaArg ", model.SYSTEM_ADMIN_ROLE_ID)
+		assert.Len(t, suggestions, 3)
+		assert.Equal(t, "this is hint 1", suggestions[0].Hint)
+		assert.Equal(t, "this is hint 2", suggestions[1].Hint)
+		assert.Equal(t, "this is hint 3", suggestions[2].Hint)
+	})
+
+	t.Run("GetAutoCompleteListItems bad arg", func(t *testing.T) {
+		suggestions := th.App.getSuggestions(emptyCmdArgs, []*model.AutocompleteData{command.AutocompleteData}, "", "bogus --badArg ", model.SYSTEM_ADMIN_ROLE_ID)
+		assert.Empty(t, suggestions)
+	})
+}
+
+type testProvider struct {
+}
+
+func (p *testProvider) GetTrigger() string {
+	return "bogus"
+}
+
+func (p *testProvider) GetCommand(a *App, T goi18n.TranslateFunc) *model.Command {
```

Comment:
not: no need to copy value here.
```suggestion
func (p *testProvider) GetCommand(_ *App, T goi18n.TranslateFunc) *model.Command {
```
We can apply this to other occurrences.

Reply:
I don't agree with this.  At minimum we would need to add this to the style guide as there doesn't seem to be a standard.  I would argue against it in that in some cases the variable name helps document the method.  For example,
```
func (m *myStruct) ImplInterface(_ []string, _ map[string]string)
```
Without seeing argument names a reader would have to figure out what interface is being implemented to look up what those arguments are for.  IMO that is not worth saving a pointer copy, especially when I would expect the compiler to do that automatically.

Reply:
I see, ideally we shouldn't see documentation from the implementation side. Also, having underscore says that this method does not uses this input. Which, I believe is good to know (I wouldn't initialize the parameter if it is not required). This is totally personal taste though, 0/5.